pkg/cmd/repo/setdefault/setdefault.go

```diff
@@ -193,7 +193,13 @@ func setDefaultRun(opts *SetDefaultOptions) error {
 			var repoNames []string
 			current := ""
 			if currentDefaultRepo != nil {
-				current = ghrepo.FullName(currentDefaultRepo)
+				// this gnarly hack covers the case where a repository has had its name
+				// or owner changed since the remotes file was originally written. See #6808
+				if currentDefaultRepo.Remote != nil {
+					if currentDefaultRepo.Remote.Resolved != "base" {
+						current = currentDefaultRepo.Remote.Resolved
+					}
+				}
```

Comment:
Do we need to make the check that the remote is not nil? Seems like that is already guaranteed by virtue of `currentDefaultRepo` being set from `ResolvedRemote()`.

Additionally if the resolved value is "base" should we be keeping the previous behavior of `current = ghrepo.FullName(currentDefaultRepo)`?

Reply:
> Do we need to make the check that the remote is not nil? Seems like that is already guaranteed by virtue of currentDefaultRepo being set from ResolvedRemote().

I suppose not but I felt like being defensive.

> Additionally if the resolved value is "base" should we be keeping the previous behavior of current = ghrepo.FullName(currentDefaultRepo)?

no; that is precisely the case that resulted in the panic of #6808

Reply:
Instead of this workaround, which is not very self-evident to me even with the added explanatory comment, could the workaround be to add a guard around `Prompter.Select` to ensure that `current` exists within `repoNames` and make the default blank if not?